File: Examples/Getting-Started/Search-Available-Jobs/Python/search_available_jobs.py
import os

from netld_example_client import NetLDClient, print_available_jobs


client = NetLDClient.from_env()
client.login()

page_data = client.search_available_jobs(
    networks=[os.environ.get("NETLD_NETWORK", "Default")],
    offset=int(os.environ.get("NETLD_JOB_OFFSET", "0")),
    page_size=int(os.environ.get("NETLD_JOB_PAGE_SIZE", "100")),
    sort_column=os.environ.get("NETLD_JOB_SORT_COLUMN", ""),
    descending=os.environ.get("NETLD_JOB_DESCENDING", "false").lower() == "true",
)

print_available_jobs(page_data)

File: Examples/Getting-Started/Search-Available-Jobs/nodeJS/search-available-jobs.mjs
import "dotenv/config";

import { NetLDClient, printAvailableJobs } from "./netld-example-client.mjs";

const client = NetLDClient.fromEnv();
await client.login();

const pageData = await client.searchAvailableJobs({
  networks: [process.env.NETLD_NETWORK || "Default"],
  offset: Number.parseInt(process.env.NETLD_JOB_OFFSET || "0", 10),
  pageSize: Number.parseInt(process.env.NETLD_JOB_PAGE_SIZE || "100", 10),
  sortColumn: process.env.NETLD_JOB_SORT_COLUMN || "",
  descending: (process.env.NETLD_JOB_DESCENDING || "false").toLowerCase() === "true",
});

printAvailableJobs(pageData);
